Demo/RCHSearchViewController.swift

```diff
@@ -52,94 +78,109 @@ class RCHSearchViewController: UIViewController, UISearchBarDelegate, UICollecti
+        let searchBuilder: RCHSearchBuilder = RCHSDK.builder(forSearch: placement, withQuery: searchTerm)
+        
+        RCHSDK.defaultClient().sendRequest(searchBuilder.build(), success: { (responseObject) in
             
-            guard let strategyResponse = responseObject as? RCHStrategyResult else {
-                print("Result Error")
+            guard let searchResult = responseObject as? RCHSearchResult else {
                 return
             }
-            guard let productArray = strategyResponse.recommendedProducts else {
-                print("Recommended Products Error")
-                return
-            }
-            self.productArray = productArray
             
-            if self.productArray.count == 0 {
-                self.showNoResults()
-            } else {
-                self.searchResultsCollectionView.isHidden = false
-                self.searchResultsCollectionView?.reloadData()
+            self.products = searchResult.products!
 
-            }
         }) { (responseObject, error) in
             print(error)
         }
-//        resetSearch()
     }
     
-    func resetSearch() {
-        view.endEditing(true)
-//        autocompleteArray = []
-        autocompleteTableView.reloadData()
-        autocompleteTableView.isHidden = true
-        searchProductsView.isHidden = false
+    func autocomplete(withQuery text: String) {
+        let autocompleteBuilder: RCHAutocompleteBuilder = RCHSDK.builderForAutocomplete(withQuery: text)
+        
+        RCHSDK.defaultClient().sendRequest(autocompleteBuilder.build(), success: { (responseObject) in
+            
+            guard let responseAutocompleteSuggestions = responseObject as? [RCHAutocompleteSuggestion] else {
+                print("Result Error")
+                return
+            }
+            
+            self.autocompleteSuggestions = responseAutocompleteSuggestions.map({$0.text!})
+        }) { (responseObject, error) in
+            print(error)
+        }
     }
     
-    func showNoResults() {
-        searchResultsCollectionView.isHidden = true
-        searchProductsView.isHidden = false
-        searchProductsLabel.text = "No Results"
-        searchProductsImageView.image = UIImage(named: "icn-tabbar-shop.pdf")
+    // MARK: IBActions
+    
+    @IBAction func handleTapOnFooter(sender: UITapGestureRecognizer) {
+        resetSearch()
     }
     
+    // MARK: UISearchBarDelegate
+    
     func searchBarTextDidBeginEditing(_ searchBar: UISearchBar) {
         searchProductsView.isHidden = true
-        autocompleteTableView.isHidden = false
     }
     
     func searchBarSearchButtonClicked(_ searchBar: UISearchBar) {
-        guard let searchTerm = searchBar.text else {
+        guard let searchText = searchBar.text else {
             print("Error: no search term entered")
             return
         }
-        searchForProducts(withTerm: searchTerm)
+        searchTerm = searchText
```

Comment:
Since we're searching as we go, I don't think we have to update the searchTerm here.